### PR TITLE
feat: add bot js template

### PR DESCRIPTION
**Pull request by IvanJobs**
To check if the code follows microsoft js code style, add bot js template for review.

### CODE REVIEW

File: bot/js/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "start": "node ./index.js",
+      "watch": "nodemon ./index.js"
+    },
+    "dependencies": {
+      "botbuilder": "~4.7.0",
+      "dotenv": "^8.0.0",
+      "express": "~4.17.1",
+      "html-entities": "^1.3.1"
+    }
+  }
```
Comment (henzhang-ms) [inline bot/js/package.json:21]: Should be formatted.

File: bot/js/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (henzhang-ms) [inline bot/js/index.js:1]: Do we scaffold a js file with MS copyright?

File: bot/js/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// Licensed under the MIT License.
+
+// index.js is used to setup and configure your bot
```
Comment (henzhang-ms) [inline bot/js/index.js:4]: maybe not need to explain some basic js knowledge?

File: bot/js/index.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+// Import required pckages
```
Comment (henzhang-ms) [inline bot/js/index.js:6]: I think we only need to tell user how to write bot app not how to write js code...

Comment (henzhang-ms) [reply bot/js/index.js:6]: maybe you can sync it with PM. This is just my understanding.

File: bot/js/index.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+// Import required bot services.
+// See https://aka.ms/bot-services to learn more about the different parts of a bot.
+const { BotFrameworkAdapter } = require('botbuilder');
+
+// Import bot definitions
+const { BotActivityHandler } = require('./botActivityHandler');
+
+// Read botFilePath and botFileSecret from .env file.
+const ENV_FILE = path.join(__dirname, '.env');
+require('dotenv').config({ path: ENV_FILE });
```
Comment (henzhang-ms) [inline bot/js/index.js:19]: .env is default env file. We don't need to specify it.

File: bot/js/index.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+// Create adapter.
+// See https://aka.ms/about-bot-adapter to learn more about adapters.
+const adapter = new BotFrameworkAdapter({
+    appId: process.env.BotId,
+    appPassword: process.env.BotPassword
+});
+
+adapter.onTurnError = async (context, error) => {
+    // This check writes out errors to console log .vs. app insights.
```
Comment (henzhang-ms) [inline bot/js/index.js:29]: Junjie had mentioned that for block comments, we should use /**/ style. please confirm it with PM.

File: bot/js/index.js (hunk 4 above)
Comment (henzhang-ms) [reply bot/js/index.js:6]: Btw, packages.

File: bot/js/botActivityHandler.js [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    TeamsActivityHandler,
+    CardFactory,
+    ActionTypes
+} = require('botbuilder');
+
+class BotActivityHandler extends TeamsActivityHandler {
+    constructor() {
+        super();
+        /* Conversation Bot */
+        /*  Teams bots are Microsoft Bot Framework bots.
```
Comment (henzhang-ms) [inline bot/js/botActivityHandler.js:16]: space number is inconsistent. Can we put them in one comment block?

File: bot/js/botActivityHandler.js [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+            attachmentLayout: 'list',
+            type: 'result',
+            attachments: [attachment]
+        };
+    
+        const response = {
+            composeExtension: result
+        };
+        return response;
+        }
```
Comment (henzhang-ms) [inline bot/js/botActivityHandler.js:170]: format

File: bot/js/botActivityHandler.js (hunk 7 above)
Comment (IvanJobs) [reply bot/js/botActivityHandler.js:16]: Sure.

File: bot/js/index.js (hunk 3 above)
Comment (IvanJobs) [reply bot/js/index.js:4]: Agree.

File: bot/js/index.js (hunk 5 above)
Comment (IvanJobs) [reply bot/js/index.js:15]: Updated.

File: bot/js/index.js (hunk 6 above)
Comment (IvanJobs) [reply bot/js/index.js:25]: Sure.

File: bot/js/package.json (hunk 1 above)
Comment (IvanJobs) [reply bot/js/package.json:20]: Updated.

File: bot/js/index.js (hunk 2 above)
Comment (IvanJobs) [reply bot/js/index.js:1]: Let me confirm this.

**Pull request merged**